mobile/src/main/java/org/openhab/habdroid/core/OpenHABVoiceService.java

```diff
@@ -174,16 +94,17 @@ private String extractVoiceCommand(Intent data) {
     }
 
 
-    private void sendItemCommand(final String itemName, final String command) {
+    private void sendItemCommand(final String itemName, final String command, Connection conn) {
         Log.d(TAG, "sendItemCommand(): itemName=" + itemName + ", command=" + command);
         try {
-            performHttpPost(itemName, command);
+            performHttpPost(itemName, command, conn);
         } catch (RuntimeException e) {
             Log.e(TAG, "Unable to encode command " + command, e);
         }
     }
 
-    private void performHttpPost(final String itemName, final String command) {
+    private void performHttpPost(final String itemName, final String command,
+                                 final Connection conn) {
         /* Call MyAsyncHttpClient on the main UI thread in order to retrieve the callbacks correctly.
          * If calling MyAsyncHttpClient directly, the following would happen:
          * (1) MyAsyncHttpClient performs the HTTP post asynchronously
```

Comment:
I think I does not understand this class enough, however: It seems to implement ContinuingIntentService in order to use onHandleIntent to get intents to it. However, doesn't ContinuingIntentService already do not stop automatically, so this isn't needed already?

Reply:
What do you mean by 'this' in your last sentence? In general, the use of IntentService here seems unneeded, because the whole point of IntentService is moving work into a background thread, which isn't needed here. The `ContinuingIntentService` seems like a gross hack, because it doesn't stop **at all** ... it just simply is kept running forever if it was started once :-/

My suggestion for an approach would be
- use a normal `Service`
- in onStartCommand, extract voice command from intent and pass it - together with the passed-in startId, to sendItemCommand
- return START_NOT_STICKY from this method
- when HTTP call is finished or exception happens, call stopSelf(startId)

That way, the service is automatically stopped whenever processing the last command finishes.

Reply:
I personally do not have a deep understanding in that, so I agree with that, as it sounds sane :) I'll change it, thanks for your input!

Reply:
Hmm, I'm curious how this whoe feature works (never used it before). The item name is recognized by the SpeechRecognizer and we then pass POST request to /rest/items/VoiceCommand with the item name as post data. However, the demo server returns a 404 with this :/ (and the whole logic of what happens once a voice command was received wasn't changed in this PR).